src/ArrayHelper.php

```diff
@@ -184,9 +184,17 @@ public static function getValue($array, $key, $default = null)
             return $array[$key];
         }
 
-        if (($pos = strrpos($key, '.')) !== false) {
-            $array = static::getValue($array, substr($key, 0, $pos), $default);
-            $key = substr($key, $pos + 1);
+        if (strpos($key, '.') !== false) {
+            $keys = explode('.', $key);
+            while (count($keys) != 0) {
+                $key = array_shift($keys);
+                if (isset($array[$key]) || array_key_exists($key, $array)) {
+                    $array = $array[$key];
+                } else {
+                    return $default;
```

Comment:
I think this syntax looks better
```
if (!isset($array[$key]) || !array_key_exists($key, $array)) {
    return $default;
}

$array = $array[$key];
```

Reply:
In your example if value is null than defaultValue will be returned. I think it will be more correct like this:
```php
if (!(isset($array[$key]) || array_key_exists($key, $array))) {
  return $default;
}

$array = $array[$key];
```

Reply:
You can simply check if the key exists. Why continue if the value is null?

```
if (!array_key_exists($key, $array)) {
  return $default;
}
```

Reply:
You can find explanation in `ArrayHelper::keyExists()` function:
```php
// Function `isset` checks key faster but skips `null`, `array_key_exists` handles this case
// http://php.net/manual/en/function.array-key-exists.php#107786
return isset($array[$key]) || array_key_exists($key, $array);
```

Reply:
Please use `array_key_exists` only. There is no need to check one condition twice.

Reply:
`isset` is a micro-optimization that is still valid for PHP 7.4. Happy-path is faster. I'd keep it.

Reply:
`isset` is a micro-optimization that is still valid in PHP 7.4 but I'm unsure if there's any benefit in having the method at all. It's array_key_exists cost vs method call cost.

Reply:
> `isset` is a micro-optimization that is still valid in PHP 7.4.

It is no longer valid: https://github.com/php/php-src/pull/3360

Reply:
Oh, that's good to know. Then change should be made removing `isset`.

Reply:
We should remove it from keyExists as well.

Reply:
Maybe I should also remove this from other parts of `getValue` function?

```php
if (is_array($array) && (isset($array[$key]) || array_key_exists($key, $array))) {
    return $array[$key];
}
```

```php
if (is_array($array)) {
    return (isset($array[$key]) || array_key_exists($key, $array)) ? $array[$key] : $default;
}
```

Reply:
Could you make it in new PR? :)

Reply:
Ok https://github.com/yiisoft/arrays/pull/19